cpp/src/gandiva/cache.cc

```diff
@@ -20,13 +20,14 @@
 
 namespace gandiva {
 
-static const int DEFAULT_CACHE_SIZE = 500;
+static const size_t DEFAULT_CACHE_SIZE = 128 * 1024 * 1024; //bytes or 256 MiB
 
-int GetCapacity() {
-  int capacity;
+size_t GetCapacity() {
+  size_t capacity;
   const char* env_cache_size = std::getenv("GANDIVA_CACHE_SIZE");
   if (env_cache_size != nullptr) {
-    capacity = std::atoi(env_cache_size);
+    capacity = std::stoul(env_cache_size);
```

Comment:
Can the `stoul` method throw an error if the user define a number that is bigger than the upper limit? Or if the `GANDIVA_CACHE_SIZE` is an invalid number?

Reply:
Yes, it will throw `std::invalid_argument` if no conversion could be performed with invalid value and `std::out_of_range` if the converted value would fall out of the range of the result type. @anthonylouisbsb

Reply:
why is this change required?